server/src/main/java/org/elasticsearch/rest/AbstractRestChannel.java

```diff
@@ -61,6 +64,15 @@ protected AbstractRestChannel(RestRequest request, boolean detailedErrorsEnabled
         this.filterPath = request.param("filter_path", null);
         this.pretty = request.paramAsBoolean("pretty", false);
         this.human = request.paramAsBoolean("human", false);
+
+        if (detailedErrorsEnabled == false) {
+            deprecationLogger.critical(
+                DeprecationCategory.API,
+                "http_detailed_errors",
+                "The JSON format of non-detailed errors will change in Elasticsearch 9.0 to match the JSON structure"
+                    + " used for detailed errors. To keep using the existing format, use the V8 REST API."
```

Comment:
We need the same deprecation warning in the v9 code too (when hitting this path in v8 compat mode).

I also don't think we can have a critical warning without any way for folks to resolve it before upgrading.

Reply:
I've changed it to a warning, and will copy these changes to the v9 PR

Reply:
This is now done